src/main/java/com/hierynomus/smbj/share/File.java

```diff
@@ -148,6 +150,16 @@ public void read(OutputStream destStream, ProgressListener progressListener) thr
         is.close();
     }
 
+    /***
+     * The function for truncate or set file length for a file
+     * @param endOfFile 64-bit signed integer in bytes, MUST be greater than or equal to 0
+     * @throws SMBApiException
+     */
+    public void setLength(long endOfFile) throws SMBApiException {
```

Comment:
This is my fault for didn't handle the checking. However, I got a slightly different point of view. As [MS-FSCC document](https://msdn.microsoft.com/en-us/library/cc232067.aspx), all the FileEndOfFileInformation information class should handle the checking for the EndOfFile. So, I think it is better to implement the checking in the information class instead of the method. Do you agree with this?